### PR TITLE
make online events more obvious

### LINKED ISSUE #2307: Make online events display more clearly or drop them
https://www.notion.so/gfsc/Kim-s-timeline-for-completion-12-3-24-d0e8ef7a03c44ba08e56d4477e3fea5d?pvs=4#580a9c9b11a3496e93322aa48fed59ac

## AC

- [ ] assess if we have online events
- [ ] if we do mark them visibly on the index view and show view

## Implementation plan

Reference Marks designs for TransDim (in comments) and add an icon and words indicating if the event is online.

We do not have a way to check where the event is ie zoom and it would be a large job to cover all possible providers. Opt for "online" so as not to promote as specific brand over others.

## out of scope for this ticket but related
- [ ] #2331
- [ ] #2332

### PR DESCRIPTION
**Pull request by aaaaargZombies**
fixes #2307 

Show an icon and "online" if there's an `online_address_id` both on index and show view

image

### CODE REVIEW

**Comment by kimadactyl** [comment]
Love this but that icon is a lil jarring :)

Can we either do it how @studiosquid did here: https://github.com/geeksforsocialchange/PlaceCal/issues/2307#issuecomment-2004259535

Or make sure that the logo is the same height as the other ones?

A globe icon could work too?

Apols if you're still working on it!

**Comment by aaaaargZombies** [comment]
> Or make sure that the logo is the same height as the other ones?

It's a font / unicode character so that's the correct size in relation to the other icons so making it fit would involve some very awkward pixel pushing. Could just use the place icon and the word "online". 

I looked at the transdim design and there's a few barriers to that design, firstly we don't have a good way of describing the online event - covering every possible location and I don't want to default to a brand name as synonym for online. The spot on the right hand side that lets you scan for online events is currently occupied by monthly events. So I guess the question there is: Do we want to de-prioritise that in favor of highlighting online events?

Let me know

**Comment by kimadactyl** [comment]
Oh I just meant - I'm happy with a circle or a globe as long as its the right height!

The icons should come out the right size if they're made properly at source.

Maybe @studiosquid would be a legend and provide a suitable icon for this to match the others?

**Comment by aaaaargZombies** [comment]
with dot?

image

Just to re-iterate the icons used are all unicode fonts not svgs so it's definitely better to pick one system rather than mix and match for consistency https://unicode-explorer.com/c/23FA

**Comment by kimadactyl** [comment]
They're not unicode - mark made them custom and I turned them into an icon font using this (i think!)

https://fontello.com/

dot looks great tho!

File: app/components/event/_event.html.erb [hunk 1]
```diff
@@ -28,6 +28,12 @@
         <span class="icon-font--date"></span>
         <%= date %>
       </div>
+      <% if online? %>
```
Comment (ivan-kocienski-gfsc) [inline app/components/event/_event.html.erb:31]: This is good but could we have the URL here?

**Review by ivan-kocienski-gfsc — CHANGES_REQUESTED** [review]
One question

**Comment by aaaaargZombies** [comment]
> They're not unicode - mark made them custom and I turned them into an icon font using this (i think!)
> 
> https://fontello.com/

hmmm  there's definitely some relation to the unicode characters, you're saying mark assigned a custom glyph to that unicode point?

```scss
.icon-font--date:before {
	content: "\e800";
} /* '' */
.icon-font--duration:before {
	content: "\e801";
} /* '' */
.icon-font--place:before {
	content: "\e802";
} /* '' */
.icon-font--repeats:before {
	content: "\e803";
} /* '' */
.icon-font--time:before {
	content: "\e804";
} /* '' */
```

[view code in context](https://github.com/geeksforsocialchange/PlaceCal/blob/41c2e443e82674c47eb7dcba725d17af3b6087ee/app/assets/stylesheets/base/icon-font.scss#L58-L72)

> dot looks great tho! maybe a tiny bit too much spacing between the dot and the text but all good

Yep I noticed that, didn't want to tidy up before we'd settled on an icon but I can do that now. Then address Ivan's point.

File: app/components/event/_event.html.erb (hunk 1 above)
Comment (aaaaargZombies) [reply app/components/event/_event.html.erb:31]: We discussed and it's handled on the partner view page in the page description.

**Review by ivan-kocienski-gfsc — APPROVED** [review]
After chatting with alfie i am happy to approve this

**Comment by studiosquid** [comment]
Sorry I'm incredibly late to this, do folks still need me to provide something suitable? The dot is definitely an improvement on the rather busy computer icon.

**Comment by kimadactyl** [comment]
> Sorry I'm incredibly late to this, do folks still need me to provide something suitable? The dot is definitely an improvement on the rather busy computer icon.

I'm happy if you're happy - moment has perhaps passed :)

**Comment by studiosquid** [comment]
Don't really know where to look to see what it currently looks like. Is the dot the finished result?

**Comment by kimadactyl** [comment]
yeah i think so!

**Comment by studiosquid** [comment]
If there is time to change the dot something a bit more in keeping would be better? (If not it's ok — but the dot on the Trans Dim mocks was more of an indicator rather than symbolising anything — it didn't live with the other icons) 

Screenshot 2024-03-25 at 16 48 31

SVGs below if needed. 
PlaceCal_Online_icon_A
PlaceCal_Online_icon_B

**Comment by kimadactyl** [comment]
Cheers @studiosquid, have noted this here: https://github.com/geeksforsocialchange/PlaceCal/issues/2332

**Pull request merged**